File: other/Solution137.java
import java.util.Arrays;

class Solution137 {
    public int singleNumber(int[] nums) {
    	Arrays.sort(nums);
    	for(int i = 0; i < nums.length; i++){
    		if(i + 1 == nums.length || nums[i] != nums[i + 1]){
    			return nums[i];
    		}
    		i = i + 2;
    	}
    	return -1;
    }
}

File: C++/Solution137.cpp
#include<iostream>
#include<vector>
#include<algorithm>
#include<unordered_map>
#include<unordered_set>
#include<stack>

using namespace std;
typedef long long ll;

class Solution137 {
public:
    int singleNumber(vector<int>& nums) {
        unordered_map<int, int> mp;
        for(auto ele: nums){
            mp[ele] ++;
        }
        for(auto ele: mp){
            if(ele.second == 1){
                return ele.first;
            }
        }
        return 0;
    }
};